From b28bf5b3aa14591963f02ef5f1a0828e30fe608c Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Tue, 19 Jul 2016 10:16:26 +0800
Subject: [PATCH] phy: rockchip-emmc: enable internal pull-down for strobe line

We enable it by default as we could see the usage of PCB layout
will not stuff this registor. For currently boards which soldered
it already, there should be no harmful.

Change-Id: Idc05c244dbaeebb1028e4828aa7a7d655899beb8
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/phy/phy-rockchip-emmc.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/phy/phy-rockchip-emmc.c b/drivers/phy/phy-rockchip-emmc.c
index 2853e0e89bc0..a184a0fe6083 100644
--- a/drivers/phy/phy-rockchip-emmc.c
+++ b/drivers/phy/phy-rockchip-emmc.c
@@ -79,6 +79,9 @@
 #define PHYCTRL_OTAPDLYENA_SHIFT	11
 #define PHYCTRL_OTAPDLYSEL_MASK		0xf
 #define PHYCTRL_OTAPDLYSEL_SHIFT	7
+#define PHYCTRL_REN_STRB_ENABLE		0x1
+#define PHYCTRL_REN_STRB_MASK		0x1
+#define PHYCTRL_REN_STRB_SHIFT		9
 
 struct rockchip_emmc_phy {
 	unsigned int	reg_offset;
@@ -203,6 +206,12 @@ static int rockchip_emmc_phy_init(struct phy *phy)
 				   PHYCTRL_OTAPDLYENA_MASK,
 				   PHYCTRL_OTAPDLYENA_SHIFT));
 
+	regmap_write(rk_phy->reg_base,
+		     rk_phy->reg_offset + GRF_EMMCPHY_CON2,
+		     HIWORD_UPDATE(PHYCTRL_REN_STRB_ENABLE,
+				   PHYCTRL_REN_STRB_MASK,
+				   PHYCTRL_REN_STRB_SHIFT));
+
 	regmap_write(rk_phy->reg_base,
 		     rk_phy->reg_offset + GRF_EMMCPHY_CON0,
 		     HIWORD_UPDATE(rk_phy->opdelay,
-- 
2.35.3

